src/backend/packages/inensus/prospect/src/Jobs/ExtractInstallations.php

```diff
@@ -0,0 +1,195 @@
+            $this->storeExtractedFile($fileName, $filePath, count($data));
+
+            Log::info('Prospect installations extraction job completed successfully!', [
+                'file' => basename($filePath),
+                'records_count' => count($data),
+            ]);
+        } catch (\Exception $e) {
+            Log::error('Error during Prospect installations extraction job: '.$e->getMessage(), [
+                'error' => $e->getMessage(),
+                'trace' => $e->getTraceAsString(),
+            ]);
+            throw $e;
+        }
+    }
+
+    /**
+     * Extract installation data from database.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function extractDataFromDatabase(): array {
+        try {
+            Log::info('Extracting device data from database...');
+
+            $query = Device::query()->with([
+                'device',
+                'person.addresses.geo',
+                'person.addresses.city.country',
+                'tokens.transaction',
+                'appliance',
+                'assetPerson',
+            ]);
+
+            $devices = $query->get();
+            Log::info('Found '.$devices->count().' devices to process');
+
+            $transformer = app(ProspectInstallationTransformer::class);
+            $installations = [];
+
+            foreach ($devices as $device) {
+                if (!$device->device()->exists()) {
+                    continue;
+                }
+
+                try {
+                    $installations[] = $transformer->transform($device);
+                } catch (\Exception $e) {
+                    Log::warning('Failed to transform device', [
+                        'device_id' => $device->id,
+                        'error' => $e->getMessage(),
+                    ]);
+                    continue;
+                }
+            }
+
+            Log::info('Successfully processed '.count($installations).' installations');
+
+            return $installations;
+        } catch (\Exception $e) {
+            Log::error('Error extracting data from database: '.$e->getMessage());
+            throw $e;
+        }
+    }
+
+    /**
+     * Generate a unique filename for the CSV file.
+     */
+    private function generateFileName(): string {
+        $timestamp = now()->toISOString();
+
+        return "prospect_{$timestamp}.csv";
+    }
+
+    /**
+     * Write CSV data to file.
+     *
+     * @param array<int, array<string, mixed>> $data
+     */
+    private function writeCsvFile(array $data, string $fileName): string {
+        try {
+            Log::info('Writing CSV file: '.$fileName);
+
+            $headers = array_keys($data[0]);
+            $csvContent = $this->arrayToCsv($data, $headers);
+
+            $companyDatabase = app(CompanyDatabase::class)->newQuery()->first();
+            $companyDatabaseName = $companyDatabase->getDatabaseName();
```

Comment:
The company database name for the specific tenant you are on is generally a bit tricky to get. This will keep using one company database for every tenant which is not what you want. 

This is what you want instead. Ps: use company id to make things uniform as other upload files use that same approach. 
```php
$user = User::query()->first();
$databaseProxy = app(DatabaseProxy::class);
$companyId = $databaseProxy->findByEmail($user->email)->getCompanyId();
 ```

Reply:
Yeah, you can just use `prospect/{$companyId}` this is consistent with how we started changing other jobs 👍